Source/Charts/Renderers/LineChartRenderer.swift

```diff
@@ -795,85 +811,63 @@ open class LineChartRenderer: LineRadarRenderer
 
     func drawGradientLine(context: CGContext, dataSet: LineChartDataSetProtocol, spline: CGPath, matrix: CGAffineTransform)
     {
-        context.saveGState()
-        defer { context.restoreGState() }
-
-        let gradientPath = spline.copy(strokingWithWidth: dataSet.lineWidth, lineCap: .butt, lineJoin: .miter, miterLimit: 10)
-        context.addPath(gradientPath)
-        context.drawPath(using: .fill)
-
-        let boundingBox = gradientPath.boundingBox
-        let gradientStart = CGPoint(x: 0, y: boundingBox.maxY)
-        let gradientEnd = CGPoint(x: 0, y: boundingBox.minY)
-        var gradientLocations = [CGFloat]()
-        var gradientColors = [CGFloat]()
-        var cRed: CGFloat = 0
-        var cGreen: CGFloat = 0
-        var cBlue: CGFloat = 0
-        var cAlpha: CGFloat = 0
-
-        //Set lower bound color
-        gradientLocations.append(0)
-        var cColor = dataSet.color(atIndex: 0)
-        if cColor.getRed(&cRed, green: &cGreen, blue: &cBlue, alpha: &cAlpha)
+        guard let gradientPositions = dataSet.gradientPositions else
         {
-            gradientColors += [cRed, cGreen, cBlue, cAlpha]
+            assertionFailure("Must set `gradientPositions if `dataSet.isDrawLineWithGradientEnabled` is true")
+            return
         }
 
-        //Set middle colors
-        guard let gradientPositions = dataSet.gradientPositions else
-        {
-            fatalError("Must set `gradientPositions if `dataSet.isDrawLineWithGradientEnabled` is true")
+        let boundingBox = spline.boundingBox
+        guard !boundingBox.isNull && !boundingBox.isInfinite && !boundingBox.isEmpty else {
```

Comment:
comma instead of ampersand. We're listing requirements, not performing Boolean logic.